weblate/static/editor/tools/search.js

```diff
@@ -0,0 +1,153 @@
+    $searchElment.parent().parent().after($searchPreview);
+
+    let debounceTimeout = null;
+
+    // Update the preview while typing with a debounce of 300ms
+    $searchElment.on("input", () => {
+      $searchPreview.show();
+      const userSearchInput = $searchElment.val();
+      const searchQuery = buildSearchQuery($searchElment);
+
+      // Clear the previous timeout to prevent the previous
+      // request since the user is still typing
+      clearTimeout(debounceTimeout);
+
+      // fetch search results but not too often
+      debounceTimeout = setTimeout(() => {
+        if (userSearchInput) {
+          $.ajax({
+            url: "/api/units/",
+            method: "GET",
+            data: { q: searchQuery },
+            success: (response) => {
+              // Clear previous search results
+              $searchPreview.html("");
+              $("#results-num").remove();
+              const results = response.results;
+              if (!results || results.length === 0) {
+                $searchPreview.text(gettext("No results found"));
+              } else {
+                showResults(results, $searchElment);
+              }
+            },
+          });
+        }
+      }, 300); // If the user stops typing for 300ms, the search results will be fetched
+    });
+
+    // Show the preview on focus
+    $searchElment.on("focus", () => {
+      if ($searchElment.val() !== "" && $searchPreview.html() !== "") {
+        $searchPreview.show();
+        $("#results-num").show();
+      }
+    });
+
+    // Close the preview on focusout, form submit, form reset, and form clear
+    $searchElment.on("focusout", () => {
+      // Hide after a delay to allow click on a result
+      setTimeout(() => {
+        $searchPreview.hide();
+        $("#results-num").hide();
+      }, 700);
+    });
+    $searchForm.on("submit", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+      $("#results-num").remove();
+    });
+    $searchForm.on("reset", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+      $("#results-num").remove();
+    });
+    $searchForm.on("clear", () => {
+      $searchPreview.html("");
+      $("#results-num").remove();
+      $searchPreview.hide();
+    });
+
+    /**
+     * Handles the search results and displays them in the preview element.
+     * @param {any} results fetched search results
+     * @param {jQuery} $searchElment The search input element
+     * @returns void
+     */
+    function showResults(results, $searchElment) {
+      // Show the number of results
+      if (results.length > 0) {
+        $searchElment.after(
+          `<h4 id="results-num">${gettext("Found: ")} ${results.length}</h4>`,
+        );
+      } else {
+        $("#results-num").remove();
+      }
+
+      for (const result of results) {
+        const key = result.context;
+        const source = result.source;
+
+        // Make the URL relative
```

Comment:
What is the reason for  making URL relative?

Reply:
the url returned contains different domain than the current one in the opened tab. Making it relative we insure it always open in the current domain not the one returned by the server. I could remove it. But I didn't want to assume in prod the url returned will be the same as the one of the page.

Reply:
Because if we left the url as is. When the user clicks it opens in inaccurate
 url (in development at least, not sure in prod)